glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt

```diff
@@ -121,17 +113,12 @@ class PingUploadWorker(context: Context, params: WorkerParameters) : Worker(cont
                         Glean.configuration
                     ).toFfi()
 
-                    if (result == Constants.UPLOAD_RESULT_RECOVERABLE) {
-                        uploadFailures++
-                    }
-
                     // Process the upload response
                     LibGleanFFI.INSTANCE.glean_process_ping_upload_response(incomingTask, result)
                 }
                 PingUploadTask.Wait -> return Result.retry()
                 PingUploadTask.Done -> return Result.success()
             }
-        } while (uploadFailures < MAX_RETRIES)
-        return Result.failure()
+        } while (true)
```

Comment:
Let's add a short comment explaining why this won't lead to an infinite loop :-) (i.e. we're taking care of this in glean-core)